tests/telemetry/test_telemetry.py

```diff
@@ -106,14 +110,20 @@ def test_telemetry_ouput(duthost, ptfhost):
 
     logger.info('start telemetry output testing')
     setup_telemetry_forpyclient(duthost)
+
     # wait till telemetry is restarted
     pytest_assert(wait_until(100, 10, duthost.is_service_fully_started, "telemetry"), "TELEMETRY not started")
     logger.info('telemetry process restarted. Now run pyclient on ptfdocker')
-    dut_ip = duthost.setup()['ansible_facts']['ansible_eth0']['ipv4']['address']
+
+    # Wait until the TCP port is open
+    dut_ip = duthost.mgmt_ip
+    wait_tcp_connection(localhost, dut_ip, TELEMETRY_PORT, timeout_s=60)
```

Comment:
I also saw error "socket.error: [Errno 111] Connection refused"  on my DUT. I wonder if 60 seconds is enough?